backend/migrations/2023-07-03-165000_heatmap/up.sql

```diff
@@ -0,0 +1,144 @@
+-- Your SQL goes here
+ALTER TABLE maps ADD COLUMN geometry GEOMETRY(POLYGON, 4326) NOT NULL;
+
+
+-- Calculate the bounding box of the map geometry.
+CREATE OR REPLACE FUNCTION calculate_bbox(map_id INTEGER)
+RETURNS TABLE(x_min REAL, y_min REAL, x_max REAL, y_max REAL) AS $$
+BEGIN
+    RETURN QUERY
+    SELECT
+        CAST(ST_XMin(bbox) AS REAL) AS x_min,
+        CAST(ST_YMin(bbox) AS REAL) AS y_min,
+        CAST(ST_XMax(bbox) AS REAL) AS x_max,
+        CAST(ST_YMax(bbox) AS REAL) AS y_max
+    FROM (
+        SELECT
+            ST_Envelope(geometry) AS bbox
+        FROM maps
+        WHERE id = map_id
+    ) AS subquery;
+END;
+$$ LANGUAGE plpgsql;
+
+
+-- Returns scores from 0-1 for each pixel of the map. It should never return values greater than 1.
+-- Values where the plant should not be placed are close to 0.
+-- Values where the plant should be placed are close to 1.
+-- The score at position x=0,y=0 is at the position x=x_min,y=y_min.
```

Comment:
I hope this explains it better.

Reply:
It is better but it could be improved: please say how exactly it needs to be moved/scaled with small pseudo code.

I created a left-over so that we can merge this: #649